Modules/cmathmodule.c

```diff
@@ -36,6 +36,15 @@ class Py_complex_protected_return_converter(CReturnConverter):
         data.return_conversion.append("""
 if (errno == EDOM) {
     PyErr_SetString(PyExc_ValueError, "math domain error");
+
+    PyObject *exc = PyErr_GetRaisedException();
```

Comment:
Note that this exception is not necessary a ValueError set above. It can be a MemoryError if allocation of string for error message failed, it can be some other error in future. It could even be a different ValueError.

So I think that the most reliable way is to create a ValueError object, set it's attribute, and only then set it by `PyErr_SetObject()`.